includes/collection/class-outbox.php

```diff
@@ -72,6 +86,62 @@ public static function add( $activity_object, $activity_type, $user_id, $content
 			return false;
 		}
 
+		self::invalidate_existing_items( $activity_object->get_id(), $activity_type, $id );
+
 		return $id;
 	}
+
+	/**
+	 * Invalidate existing outbox items with the same activity type and object ID
+	 * by setting their status to 'publish'.
+	 *
+	 * @param string $object_id     The ID of the activity object.
+	 * @param string $activity_type The type of the activity.
+	 * @param int    $current_id    The ID of the current outbox item to exclude.
+	 *
+	 * @return void
+	 */
+	private static function invalidate_existing_items( $object_id, $activity_type, $current_id ) {
+		$meta_query = array(
+			array(
+				'key'   => '_activitypub_object_id',
+				'value' => $object_id,
```

Comment:
For Blog users, the object_id contains a timestamp (`http://localhost:8888/?p=1834#activity-update-2025-02-11T21:02:54Z`), so there won't be a shared object_id between activities. Should we remove the hash for this purpose?

Reply:
I like the original `$wp_object`, which it already saves currently, just with the added hash. we could strip the hash in Outbox::add()

Reply:
This feels hacky! the hash was a workaround because we had no persistant IDs. I would like to fix that, so that we do no longer have the hashes anywhere.

Reply:
Sure! Could I leave that with you? I still don't feel super confident in that area

Reply:
Should work now!